crates/volta-core/src/error/mod.rs

```diff
@@ -67,6 +81,83 @@ impl From<ErrorKind> for VoltaError {
     }
 }
 
+pub trait AcceptableErrorToValue<T> {
+    fn accept_error_as_value_if<F1, F2>(self, accept_if: F1, as_value: F2) -> Fallible<T>
+    where
+        F1: FnOnce(&VoltaError) -> bool,
+        F2: FnOnce() -> T;
+}
+
+pub trait AcceptableErrorToDefault<T>: AcceptableErrorToValue<T>
```

Comment:
I haven't worked all the way through it, but I think a lot of this machinery could be handled with a helper function and the built-in `Result::or_else` method, which I think would simplify it a fair amount.